index.bs

```diff
@@ -357,59 +356,102 @@ The client facilitates these security measures by providing correct origins and
-            optional AssertionOptions       options
-        );
     };
 </xmp>
+<dl dfn-type="attribute" dfn-for="ScopedCredential">
+    :   {{Credential/id}}
+    ::  {{ScopedCredential}} overrides {{Credential}}'s getter for this attribute, returning the [=base64url encoding=] of the
+        data contained in the object's {{ScopedCredential/[[identifier]]}} slot.
+
+    :   {{ScopedCredential/rawID}}
+    ::  This attribute returns the {{ArrayBuffer}} contained in the {{ScopedCredential/[[identifier]]}} internal slot.
+
+    :   <dfn>response</dfn>
+    ::  This attribute contains the [=authenticator=]'s response to the client's assertion or attestation request. If the
+        {{ScopedCredential}} is created in response to {{ScopedCredential/create()}}, this attribute's
+        value will be an {{AuthenticatorAttestationResponse}}, otherwise, the {{ScopedCredential}} was created in response to
+        {{CredentialsContainer/get()}}, and this attribute's value will be an {{AuthenticatorAssertionResponse}}.
+
+    :   <dfn>\[[type]]</dfn>
+    ::  The {{ScopedCredential}} [=interface object=]'s {{Credential/[[type]]}} slot's value is the string "`scoped`".
+
+        Note: This is reflected via the {{Credential/type}} attribute getter inherited from {{Credential}}.
+
+    :   <dfn>\[[identifier]]</dfn>
+    ::  This internal slot contains an identifier for the credential, chosen by the platform with help from the authenticator.
+        This identifier is used to look up credentials for use, and is therefore expected to be globally unique with high
+        probability across all credentials of the same type, across all authenticators. This API does not constrain the format
+        or length of this identifier, except that it must be sufficient for the platform to uniquely select a key. For example,
+        an authenticator without on-board storage may create identifiers that consist of the key material wrapped with a key
+        that is burned into the authenticator.
+
+    :   <dfn method>create(accountInformation, cryptoParameters, attestationChallenge, options)</dfn>
+    ::  This method allows a developer to request the User Agent to create a new credential, and persist it to the underlying
+        platform, which may involve data storage managed by the browser or the OS. The user agent will prompt the user to
+        approve this operation. On success, the promise will be resolved with a {{ScopedCredential}} which contains an
+        {{AuthenticatorAttestationResponse}} object. Implementation details are found in [#createCredential].
+
+    :   <dfn method>\[[Retrieve]](options, mediation)</dfn>
+    ::  The {{ScopedCredential}} [=interface object=] defines this internal method to override {{Credential}}'s implementation,
+        which will enable support for assertion generation through {{CredentialsContainer/get()}}. Implementation details are
+        found in [[#getAssertion]].
```

Comment:
> How does this actually show up in the author's code?

It doesn't. The `[[...]]` syntax is what the ECMAScript spec uses for internal properties... slots and methods that exist on an object for explanation purposes, but don't have IDL bindings).

> We're saying here that a method in an internal slot of the ScopedCredential object overrides a method of the CredentialsContainer interface.

No, sorry for the confusion: here, we're overriding `Credential`'s implementation of `[[Retreive]]`. `[[Retrieve]]` is _used_ by `CredentialsContainer/get()`, but isn't monkeying around with `CredentialsContainer`'s implementation. I'll walk through the new flow below:

> Does this then constitute a monkey patch on section 4.1.1 of Credentials Management?

In https://w3c.github.io/webappsec-credential-management/base.html, I'm fiddling around with how `get()` is wired up to specific kinds of credential objects. The existing document is pretty hand-wavey about how it knows that `get({ 'password': true })` is a request that operates on `PasswordCredential` objects. That new document attempts to be a little more rigorous. I'll try to spell it out briefly here:

1.  All `Credential` interface objects have an internal (e.g. not exposed to the web) implementation of `[[Retrieve]](options, mediation)` (https://w3c.github.io/webappsec-credential-management/base.html#credential-internal-methods). 

2.  During `get()`, we walk through all the interface objects that derive from `Credential` directly or indirectly (step 5.1 of https://w3c.github.io/webappsec-credential-management/base.html#abstract-opdef-request-a-credential), and examine their `[[type]]` slot's value. If that value is the same as one of the keys in the dictionary that we passed into `get()`, then we'll grab the interface object's `[[Retrieve]]` method.

3.  If, after walking through all such objects, we've only gotten one `[[Retrieve]]` method, then we'll execute it and return the resulting `Credential` to the developer.

    If we got more than one hit (e.g. the developer passed in `{ 'password': true, 'scoped': { ... } }`), then we'll error out.

It might be easier to explain this on a call. It's a bit convoluted, but I do think it's more clearly explained than the last attempt, at least from a mechanical perspective of what the browser would actually do. :)